apis/common/v1/connection_details.go

```diff
@@ -54,6 +60,22 @@ type ConnectionSecretMetadata struct {
 	Type *corev1.SecretType `json:"type,omitempty"`
 }
 
+// SetOwnerUID sets owner object uid label.
+func (in *ConnectionSecretMetadata) SetOwnerUID(so metav1.Object) {
+	if in.Labels == nil {
+		in.Labels = map[string]string{}
+	}
+	in.Labels[LabelKeyOwnerUID] = string(so.GetUID())
+}
```

Comment:
Presuming this is used to track ownership, should we add a note on the `ConnectionSecretMetadata` type that Secret Store implementations must implement label support in order to track secret ownership?